docs/config.rst

```diff
@@ -11,7 +11,7 @@ An example of a ``gordon-janitor.toml`` file:
     :language: ini
 
 
-You may choose to have a ``gordon-janitor-user.toml`` file for development. Any top-level key will override what's found in ``gordon-janitor.toml``.
+You may choose to have a ``gordon-janitor-user.toml`` file for development. All tables are deep merged into ``gordon-janitor.toml``, to limit the amount of config duplication needed. For example, you can reset ``core.debug`` without having to redeclare which plugins you'd like to run).
```

Comment:
I'm not sure it's clear what 'deep merged' means.  'Recursively merged, with user keys overriding global ones'?

Reply:
When I was creating the initial PR I chose deep with the analogy to deep copy. Googling around, it seems like "deep merge" is more common than "recursively merge", at least with Python dictionaries. 

I have updated the wording to include "override" in the example. Does that make it more clear?